alertmanager/alertmanager.go

```diff
@@ -0,0 +1,203 @@
+package alertmanager
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/prometheus/alertmanager/api"
+	"github.com/prometheus/alertmanager/config"
+	"github.com/prometheus/alertmanager/dispatch"
+	"github.com/prometheus/alertmanager/inhibit"
+	"github.com/prometheus/alertmanager/nflog"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/provider/mem"
+	"github.com/prometheus/alertmanager/silence"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/alertmanager/ui"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/route"
+	"github.com/weaveworks/mesh"
+)
+
+// Config configures an Alertmanager.
+type Config struct {
+	UserID      string
+	DataDir     string
```

Comment:
Oh yeah, that's another (big?) followup thing to figure out - right now this will just persist silences and notification logs onto local disk into this directory, as the "normal" Alertmanager does. If we run AM in HA mode, can we mount in per-replica volumes into each one? Otherwise we'll have to do deeper surgery on AM packages to allow for non-local-file storage. Deferring this until later though.